src/v/cluster/topic_table.cc

```diff
@@ -378,9 +378,6 @@ topic_table::apply(create_non_replicable_topic_cmd cmd, model::offset o) {
 }
 
 void topic_table::notify_waiters() {
-    if (_waiters.empty()) {
```

Comment:
this is not correct, this way when there are no waiters, they will loose `_pending_changes` since they are swapped on 385. The logic behind this piece was to waiters never lose deltas.

Reply:
Maybe we can keep a piece of state to control notification logic f.e. last notified offset. This way we will keep the condition waiting for waiters, but we will notify earlier with only those entries which weren't yet seen by notification listener.

Reply:
Makes sense, explains CI too

Reply:
Its weird because this also doesn't work if there is more then one waiter, and the API seems like its ok to have multiple waiters, thats why i used the notification system in my changes in the first place. Maybe we should file this under technical debt too and give it some TLC